Update_db_columns_for_tags

navjot-nangia opened the pull request:
Add first and final tag columns

I used refine_missing_red branch because it already had some changes that were needed for these commits. 
- updated methods in mask_inspection.py for better readability
- changed name of columns for voxel tags in db
- the final tag is "good" and it will be in column "final_voxel_tag"
- added functionality to not refine images if the "final_voxel_tag" is good

sbaghba committed:
creating PR draft

sbaghba committed:
Added base preprocess mode (no split)

sbaghba committed:
previous commits mistakenly reverted

sbaghba committed:
added remove src and empty removal

mkutu committed:
minor fix for filtering by tags

mkutu committed:
exr and quick fixex to inspect

mkutu committed:
Quick fix in mossing_red

sbaghba committed:
seperated the utility files for preprocessing

sbaghba committed:
added multi-processing

navjot-nangia committed:
- Updated methods in mask_inspection.py for better readability and consistency.
- changed name of columns for voxel tags in db
- the final tag is "good" and it will be in column "final_voxel_tag"
- added functionality to not refine images if the "final_voxel_tag" is good

navjot-nangia committed:
- Updated methods in mask_inspection.py for better readability and consistency.
- changed name of columns for voxel tags in db
- the final tag is "good" and it will be in column "final_voxel_tag"
- added functionality to not refine images if the "final_voxel_tag" is good

navjot-nangia committed:
minor change

navjot-nangia committed:
minor change

navjot-nangia committed:
Update docstring to clarify

navjot-nangia committed:
Update docstring to clarify

mkutu commented on src/mask_gen_utils/mask_refine.py:
```diff
@@ -94,13 +94,16 @@ def _map_voxel_tags(self) -> Dict[str, str]:
         tag_map = {}
         for tag_label, keyword in tag_keywords.items():
-            matched = self.df_voxel_db[self.df_voxel_db["voxel tags"].str.contains(keyword, na=False)]["image_name"]
+            # filter the DataFrame for images that are not tagged "good"
```
this method should only be mapping user defined tags to their canonical version. Let's separate any filtering out of "good" tags into a separate method

navjot-nangia replied:
- added filter_db_by_final_voxel_tags in conf to track which tags to filter for from final_voxel_tags
- separated out method to get df that needs refinement, aka, filter out rows with tags in filter_db_by_final_voxel_tags
- separated out matched and unmatched voxel tag map
- added a few lines of docs in init

mkutu replied:
this is still in the same method. This isn't what I meant. You need separate different steps, not just nest them into another method

navjot-nangia replied:
But the filtering of df for good tags is being done in a different step:  _get_df_needing_refinement

mkutu replied:
we need to create a pipeline with one task occuring after the other. This is more a maze of nested tasks that are hard to follow

mkutu committed:
Refactor database handling in FiftyOneMaskInspector for improved clarity and functionality

mkutu committed:
Refactor database handling in FiftyOneMaskInspector for improved clarity and functionality

sbaghba committed:
implemented datastats calculation

navjot-nangia committed:
- added filter_db_by_final_voxel_tags in conf to track which tags to filter for from final_voxel_tags
- separated out method to get df that needs refinement, aka, filter out rows with tags in filter_db_by_final_voxel_tags
- separated out matched and unmatched voxel tag map
- added a few lines of docs in init

navjot-nangia committed:
- added filter_db_by_final_voxel_tags in conf to track which tags to filter for from final_voxel_tags
- separated out method to get df that needs refinement, aka, filter out rows with tags in filter_db_by_final_voxel_tags
- separated out matched and unmatched voxel tag map
- added a few lines of docs in init

navjot-nangia committed:
minor

navjot-nangia committed:
minor

sbaghba committed:
added normalization to dataset and set the flag in train config

navjot-nangia commented on src/mask_gen_utils/mask_refine.py:
```diff
@@ -84,26 +91,62 @@ def _load_voxel_db(self) -> pd.DataFrame:
-        """
-        Maps voxel inspection tags to canonical tags defined in the configuration.
+
+    def _get_df_needing_refinement(self) -> pd.DataFrame:
```
@mkutu 
Here:

 the filtering of df for good tags is being done in a different step: _get_df_needing_refinement

sbaghba committed:
data_stats

mkutu committed:
Merge branch 'develop' into HPreprocess

mkutu committed:
Update default.yaml

minor name correct

mkutu committed:
Update default.yaml

remove wanb

mkutu committed:
Update main.py

mkutu committed:
Merge pull request #79 from precision-sustainable-ag/HPreprocess

Preprocess PR

navjot-nangia commented on src/mask_gen_utils/mask_inspect.py:
```diff
@@ -164,75 +184,121 @@ def create_dataset(self, samples) -> fo.Dataset:
+        if new_rows:
+            self.df = pd.concat([self.df, pd.DataFrame(new_rows)], ignore_index=True)
+
+    def _populate_empty_db(self, current_data: dict) -> None:
```
The rows of self.df were not populating if new db needed to be created. So, I changed this methods and now it is working.

navjot-nangia committed:
Refactor _populate_empty_db to make sure the db is being created

navjot-nangia committed:
-refactored self._get_df_needing_refinement to add it as a step in mask_refine.py
- removed unnecessary nesting of methods

mkutu commented on src/mask_gen_utils/mask_inspect.py:
```diff
@@ -164,75 +184,125 @@ def create_dataset(self, samples) -> fo.Dataset:
             fo.delete_dataset(self.dataset_name)
```
@navjot-nangia why are we deleting the dataset if it already exists? This would probably solve a lot of our db updating.

mkutu committed:
major refactor of inspect: use sql db instead of csv

mkutu committed:
Refactor mask refinement process to utilize configuration for filtering tags and improve tag mapping logic

mkutu reviewed (changes requested):
this needs refactoring

mkutu committed:
refactor mask_inspect.py

mkutu committed:
Refactor mask refinement, use sql db

mkutu committed:
tag filtering and get images for refinement

mkutu commented:
TODO:
1. ~~add processing parameters to DB so that we  know what was used to process each image~~
2. ~~store and access DB from LTS (setup configs and classes)~~
3. ~~Separate the DB class in it's own utils file~~


attn: @navjot-nangia

mkutu committed:
update gitignore

navjot-nangia commented:
@mkutu I will work on these

mkutu commented:
> @mkutu I will work on these 

Let me work on #2

navjot-nangia commented:
Ok 👍🏽

mkutu committed:
updated DB location and table usage

mkutu committed:
update database file name for agir_field_db to test location

mkutu committed:
Added refine parameters to db

mkutu committed:
separated db operations into db_utils submodule
